File: hanghae-cinema/src/main/java/com/example/hanghaecinema/domain/Movie.java
package com.example.hanghaecinema.domain;

import com.example.hanghaecinema.domain.enums.AgeRating;
import com.example.hanghaecinema.domain.enums.Genre;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@Entity
@Table(name = "movies")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Movie extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String title;

    @Enumerated(value = EnumType.STRING)
    private Genre genre;

    private LocalDate releaseDate;

    @Column(nullable = false)
    private Integer runtimeMinutes;

    @Enumerated(value = EnumType.STRING)
    private AgeRating ageRating;

    @Column(length = 500)
    private String thumbnailUrl;

}

File: cinema/cinema-domain/src/main/kotlin/com/study/cinema/domain/movie/Movie.kt
package com.study.cinema.domain.movie

import jakarta.persistence.Column
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import org.hibernate.annotations.Comment
import java.time.LocalDate

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Movie(
    title: String,
    genre: Genre,
    movieRating: MovieRating,
    releaseDate: LocalDate,
    runningTimeMinutes: Int,
    posterUrl: String,
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0

    @Comment("제목")
    var title: String = title
        protected set

    @Comment("장르")
    @Enumerated(EnumType.STRING)
    @Column(length = 50)
    var genre: Genre = genre
        protected set

    @Comment("영상물 등급")
    @Enumerated(EnumType.STRING)
    @Column(length = 50)
    var movieRating: MovieRating = movieRating
        protected set

    @Comment("개봉일")
    var releaseDate: LocalDate = releaseDate
        protected set

    @Comment("러닝 타임 (분)")
    var runningTimeMinutes: Int = runningTimeMinutes
        protected set

    @Comment("포스터 URL")
    var thumbnailUrl: String = posterUrl
        protected set
}